### DISCUSSION TITLE
How to use addTextToActiveTabEditor(text) function?

### DISCUSSION THREAD

**Discussion by ElleryQu** [TriliumNext/Trilium#1506] [Q&A]
Hello, i wanna insert a time stamp text after the cursor, but i get failed. What's wrong with it?
`api.addButtonToToolbar({
    title: 'addTimeStamp',
    icon: 'check',
    shortcut: 'ctrl+g',
    action: async function() {
        await api.addTextToActiveTabEditor(new Date().getHours()+":"+new Date().getMinutes()+" "）;
        await api.waitUntilSynced()
    }
});`
  

In fact, i can't find the function's definition(addTextToActiveTabEditor(text)) in following files, which i thought perhaps related to:
1. app_context.js, 2. entrypoints.js, 3. tabmanager.js, 4. component.js.

**Comment by zadam** [comment]
Hi, this is actually a built-in function, try pressing `ALT-T` to insert timestamp.

I'm not sure what's the problem in your script but this one is working:

```javascript
api.addButtonToToolbar({ 
    title: 'addTimeStamp', 
    icon: 'check', 
    shortcut: 'ctrl+g', 
    action: async function() {
        await api.addTextToActiveTabEditor(new Date().getHours()+":"+new Date().getMinutes()+" ");
    }
});
```

**Reply by ElleryQu** [reply]
Thanks for your reply! It may be because i'm using the version 0.43.
